### PR TITLE
A few things updated into dev branch, took on board changes and reworked code to flow in line with re-write

### PR DESCRIPTION
**Pull request by ZephrFish**
New:
- Added AzVault module for spraying
- Added MS_Template for creating microsoft/azure specific spraying modules

Updated:
- Added in discord webhook function into notify
- Pushed fire.py into utils folder for tidier root, changed credmaster.py to import from correct location

WIP:
- Began work on rewrite of config file to factor in other webhooks
- Began research into GCP/Azure alternatives for API gateway and increased opsec

### CODE REVIEW

**Comment by ZephrFish** [comment]
Took on board all comments from previous pull and corrected areas where required and added in new additions to align to re-write

File: plugins/azvault/azvault.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+    spoofed_ip = utils.generate_ip()
+    amazon_id = utils.generate_id()
+    trace_id = utils.generate_trace_id()
+
+    headers = {
+        "X-My-X-Forwarded-For" : spoofed_ip,
+        "x-amzn-apigateway-api-id" : amazon_id,
+        "X-My-X-Amzn-Trace-Id" : trace_id,
+        "User-Agent" : useragent,
```
Comment (knavesec) [inline plugins/azvault/azvault.py:30]: In previous PR, you noted that this UA needed to be static to evade defender for cloud or something like that?

File: plugins/template/MS_Template/template.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+    spoofed_ip = utils.generate_ip()
+    amazon_id = utils.generate_id()
+    trace_id = utils.generate_trace_id()
+
+    headers = {
+        "X-My-X-Forwarded-For" : spoofed_ip,
+        "x-amzn-apigateway-api-id" : amazon_id,
+        "X-My-X-Amzn-Trace-Id" : trace_id,
+        "User-Agent" : useragent,
```
Comment (knavesec) [inline plugins/template/MS_Template/template.py:30]: Same question as above, does the UA matter?

Comment (ZephrFish) [reply plugins/template/MS_Template/template.py:30]: It doesn't in this instance as I tested it against a few endpoints and it doesn't flag :)

File: plugins/azvault/azvault.py (hunk 1 above)
Comment (ZephrFish) [reply plugins/azvault/azvault.py:30]: It doesn't in this instance as I tested it against a few endpoints and it doesn't flag :)

**Pull request merged**